packages/admin/admin/src/form/FinalFormChooseFile.tsx

```diff
@@ -0,0 +1,269 @@
+        files = [fieldValue];
+    } else {
+        files = [];
+    }
+
+    const rejectedFiles = fileRejections.map((rejectedFile) => (
+        <div key={rejectedFile.file.name} className={classes.rejectedFileListItem}>
+            <div className={classes.fileListText}>{rejectedFile.file.name}</div>
+            <IconButton color="error">{infoIcon}</IconButton>
+        </div>
+    ));
+
+    return (
+        <div className={classes.root}>
+            <div {...getRootProps()} className={classes.dropzone}>
+                <input {...getInputProps()} />
+                {!disableDropzone && (
+                    <div
+                        className={clsx(
+                            classes.droppableArea,
+                            disabled && classes.droppableAreaIsDisabled,
+                            isDragReject && classes.droppableAreaHasError,
+                        )}
+                    >
+                        {isDragReject && <div className={classes.droppableAreaError}>{infoIcon}</div>}
+                        <div className={classes.droppableAreaCaption}>
+                            <FormattedMessage id="comet.finalFormChooseFile.dropfiles" defaultMessage="Drop files here to upload" />
+                        </div>
+                    </div>
+                )}
+                {!disableSelectFileButton && (
+                    <Button disabled={disabled} variant="contained" color="primary" startIcon={selectIcon}>
+                        <FormattedMessage id="comet.finalFormChooseFile.selectfile" defaultMessage="Select file" />
+                    </Button>
+                )}
+            </div>
+            {files.length > 0 && (
+                <div className={classes.fileList}>
+                    {files.map((file) => (
+                        <div key={file.name} className={classes.fileListItem}>
+                            <div className={classes.fileListText}>{file.name}</div>
+                            <div className={classes.fileListItemInfos}>
+                                <Chip label={<PrettyBytes value={file.size} />} />
+                                <IconButton onClick={removeFile(file)}>{deleteIcon}</IconButton>
+                            </div>
+                        </div>
+                    ))}
+                </div>
+            )}
+            {fileRejections.length > 0 && <div className={classes.fileList}>{rejectedFiles}</div>}
+            {(fileRejections.length > 0 || isDragReject) && (
+                <div className={classes.errorMessage}>
+                    <Info color="error" />
+                    <FormattedMessage id="comet.finalFormChooseFile.errors.unknownError" defaultMessage="Something went wrong." />
+                </div>
+            )}
+            <FormHelperText>
+                <FormattedMessage id="comet.finalFormChooseFile.maximumFileSize" defaultMessage="Maximum file size" /> <PrettyBytes value={maxSize} />
```

Comment:
Maybe make the file size part of the translatable string, incase some languages require that value to be in a different part of the text instead of at the end. 

```suggestion
                <FormattedMessage
                    id="comet.finalFormChooseFile.maximumFileSize"
                    defaultMessage="Maximum file size {fileSize}"
                    values={{
                        fileSize: <PrettyBytes value={maxSize} />,
                    }}
                />
```